Sources/Alchemy/SQL/QueryBuilder/Query.swift

```diff
@@ -208,30 +369,64 @@ public class Query: Sequelizable {
         return self
     }
 
+    /// A helper for adding an **or** `whereNull` clause.
+    ///
+    /// - Parameters:
+    ///   - key: The column to match against
+    /// - Returns: The current query builder `Query` to chain future queries to
     public func orWhereNull(key: String) -> Self {
         return self.whereNull(key: key, boolean: .or)
     }
 
+    /// Add a where clause requiring that a column not be null.
+    ///
+    /// - Parameters:
+    ///   - key: The column to match against
+    ///   - boolean: How the clause should be appended (*and* or *or*)
+    /// - Returns: The current query builder `Query` to chain future queries to
     public func whereNotNull(key: String, boolean: WhereBoolean = .and) -> Self {
         return self.whereNull(key: key, boolean: boolean, not: true)
     }
 
+    /// A helper for adding an **or** `whereNotNull` clause.
+    ///
+    /// - Parameters:
+    ///   - key: The column to match against
+    /// - Returns: The current query builder `Query` to chain future queries to
     public func orWhereNotNull(key: String) -> Self {
         return self.whereNotNull(key: key, boolean: .or)
     }
 
-
+    /// Add a having clause to filter results from aggregate functions.
+    ///
+    /// - Parameters:
+    ///   - clause: A `WhereValue` clause matching a column to a value
+    /// - Returns: The current query builder `Query` to chain future queries to
     public func having(_ clause: WhereValue) -> Self {
```

Comment:
Could leverage another type like `AggregateQuery` or something that these functions could be on to help avoid inevitable errors if this gets chained onto a non-aggregate query. 

No need to bother now since it's a tiny improvement but just thinking out loud; would help leverage Swift type system to avoid SQL errors... The kind of benefits that would be unique to a Swift web framework.